feat: migrate portfolio-review-step

chelaz1234 opened the pull request:
_Any italic text should be deleted from the final Pull Request text, including this line_

# Description

This component is allows students to review their selected tasks and learning summary report before final submission. It checks for required components like task selection and ILO alignment, displays warnings if anything is missing, and lets students confirm their portfolio is ready for assessment._

## Type of change

- Migrated: portfolio-review-step.coffee →portfolio-review-step.component.ts, .html, .scss

- Deleted: portfolio-review-step.coffee,portfolio-review-step.tpl.html

- Updated: doubtfire-angularjs.module.ts (removed old directive reference)

- Updated: portfolio.tpl.html to mount the new Angular 17 component

# How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_

Before:
<img width="1918" height="1027" alt="portfiloio ss" src="https://github.com/user-attachments/assets/48288546-6dc3-481b-a295-52bbaf525c75" />

After:
<img width="1074" height="853" alt="Screenshot 2025-09-11 200938" src="https://github.com/user-attachments/assets/c548330d-c193-46cb-ae6c-e9059816781d" />
<img width="1920" height="965" alt="449666951-542a0b80-550f-4432-ae4d-68d3fb882dc5" src="https://github.com/user-attachments/assets/d92e2f64-18a5-441c-880a-4f7d586ebe68" />


# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

chelaz1234 committed:
feat: migrate portfolio-review-step

chelaz1234 committed:
Update portfolio-review-step.component.ts

chelaz1234 committed:
Update and rename portfolio-review-step.component.css to portfolio-review-step.component.scss

chelaz1234 committed:
Update portfolio-review-step.component.ts

chelaz1234 committed:
Update portfolio-review-step.component.html

Triet-coder reviewed (approved):
<img width="1279" height="356" alt="Screenshot 2025-09-21 143210" src="https://github.com/user-attachments/assets/90d5e75d-d736-4201-b1ca-a9c094483705" />
<img width="298" height="92" alt="Screenshot 2025-09-21 143114" src="https://github.com/user-attachments/assets/ca61d5e1-1ca7-4946-814d-f231f3bb43ba" />
Solid migration — review step validates task selection + ILO, shows clear warnings, and the confirm flow is clean. UI feels responsive; no regressions. Tiny nits: add a test for missing-state edge cases and disable Confirm until all checks pass.

EkamBhullar commented:
Looks great. I have tested this migration, and it works perfectly fine. You can just try including some screenshots, so it is easy to compare

chelaz1234 commented:
Hi @EkamBhullar, I’ve added screenshots showing the before and after states of the migration. You can now compare them side by side.

chelaz1234 commented:
> <img alt="Screenshot 2025-09-21 143210" width="1279" height="356" src="https://private-user-images.githubusercontent.com/177189697/492011430-90d5e75d-d736-4201-b1ca-a9c094483705.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg5NDMyNDUsIm5iZiI6MTc1ODk0Mjk0NSwicGF0aCI6Ii8xNzcxODk2OTcvNDkyMDExNDMwLTkwZDVlNzVkLWQ3MzYtNDIwMS1iMWNhLWE5YzA5NDQ4MzcwNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyN1QwMzE1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MjRkNWM4ZDczOWEzMjg0OWMzOTQzODNkYzk1ZGVmMTAwMWRiZjZiM2ZhMjZhYjQ5MGMzMzAzNzA3OGRhZTdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3H7LDskNDYm5Zpy6ac1pHWR8f_1_kW1Dmvjjb1HY5Xs"> <img alt="Screenshot 2025-09-21 143114" width="298" height="92" src="https://private-user-images.githubusercontent.com/177189697/492011438-ca61d5e1-1ca7-4946-814d-f231f3bb43ba.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg5NDMyNDUsIm5iZiI6MTc1ODk0Mjk0NSwicGF0aCI6Ii8xNzcxODk2OTcvNDkyMDExNDM4LWNhNjFkNWUxLTFjYTctNDk0Ni04MTRkLWYyMzFmM2JiNDNiYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkyN1QwMzE1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTUzZDdlMjI4NmE1OTNlNzMxYmQxYTRlZWU1MjMyYmIxNjhiOGVlZGNmN2YxMzE3YTViYzFmNTU2YWU2NDEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WFVOv0Z9_bkroGD0einS4CkJ8D1yOd3yiQnIbxloALA"> Solid migration — review step validates task selection + ILO, shows clear warnings, and the confirm flow is clean. UI feels responsive; no regressions. Tiny nits: add a test for missing-state edge cases and disable Confirm until all checks pass.

hanks for the detailed review! 
I’ve fixed the suggestions you mentioned — added a test for the missing-state edge cases and updated the flow to keep the Confirm button disabled until all checks pass.

HasinduWelarathne reviewed (approved):
Tested locally. All the functions behaves correctly , routing/deep links OK, no regressions observed. Code looks clean. LGTM
<img width="1366" height="689" alt="Capture after migration" src="https://github.com/user-attachments/assets/5c708582-ad27-498c-a277-544e14c8abe6" />
